src/node_wasi.cc

```diff
@@ -55,13 +60,15 @@ inline void Debug(WASI* wasi, Args&&... args) {
     (result) = js_value->type ## Value(&lossless);                            \
   } while (0)
 
-#define GET_BACKING_STORE_OR_RETURN(wasi, args, mem_ptr, mem_size)            \
-  do {                                                                        \
-    uvwasi_errno_t err = (wasi)->backingStore((mem_ptr), (mem_size));         \
-    if (err != UVWASI_ESUCCESS) {                                             \
-      (args).GetReturnValue().Set(err);                                       \
-      return;                                                                 \
-    }                                                                         \
+#define GET_BACKING_STORE_OR_RETURN(wasi, args, mem_ptr, mem_size)             \
+  do {                                                                         \
+    Local<WasmMemoryObject> _memory =                                          \
+        PersistentToLocal::Strong((wasi)->memory_);                            \
+    std::shared_ptr<BackingStore> _backing_store =                             \
+        _memory->Buffer()->GetBackingStore();                                  \
+    *(mem_size) = _backing_store->ByteLength();                                \
+    *(mem_ptr) = static_cast<char*>(_backing_store->Data());                   \
+    CHECK(mem_ptr != nullptr);                                                 \
```

Comment:
This looks neither safe nor sound: you keep the pointer but let the shared_ptr go out of scope at the end of the block.

Reply:
yep yep this will be deleted once all the functions are migrated, i just am keeping it around so i can compile in the intermediate